include/swift/AST/DiagnosticsFrontend.def

```diff
@@ -71,6 +71,10 @@ ERROR(error_option_requires_sanitizer, none,
       "option '%0' requires a sanitizer to be enabled. Use -sanitize= to "
       "enable a sanitizer", (StringRef))
 
+ERROR(error_option_requires_specific_sanitizer, none,
+      "option '%0' requires '%1' sanitizer to be enabled. Use -sanitize=%1 to "
+      "enable the sanitizer", (StringRef, StringRef))
```

Comment:
Two reasons

1. Using `-sanitize-recover=address` without `-sanitize=address` is a mistake in the build system and we should catch that.
2. Clang doesn't automatically enable a sanitizer when provided `-fsanitize-recover=address` either. It doesn't produce an error though, but I think producing an error here is better.

Reply:
@brentdax I've actually changed my mind here. After thinking about this some more I realised there's a legitimate use case where passing `-sanitize-recover=address` without `-sanitize=address` should not result in an error. Consider a scenario where `-sanitize-recover=address` is globally passed to the build of a project but `-sanitize-recover=address` is not enabled everywhere in the project (e.g. project is being incrementally ported to support ASan). The current version of this PR would cause a compiler error which is unhelpful. Instead I'm going to make this a warning.

Reply:
I've made this a warning now.